x-pack/platform/plugins/shared/stack_connectors/public/common/auth/header_fields.test.tsx

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { render, screen, waitFor } from '@testing-library/react';
+import { AuthFormTestProvider } from '../../connector_types/lib/test_utils';
+import userEvent from '@testing-library/user-event';
+
+import { HeaderFields } from './header_fields';
+
+describe('HeaderFields', () => {
+  const onSubmit = jest.fn();
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('renders one default header row', async () => {
+    render(
+      <AuthFormTestProvider onSubmit={onSubmit}>
+        <HeaderFields maxHeaders={20} readOnly={false} />
+      </AuthFormTestProvider>
+    );
+
+    expect(await screen.findByText('HTTP headers')).toBeInTheDocument();
+
+    expect(screen.getByTestId('webhookHeadersKeyInput')).toBeInTheDocument();
+    expect(screen.getByTestId('webhookHeadersValueInput')).toBeInTheDocument();
+    expect(screen.getByTestId('webhookHeaderTypeSelect')).toBeInTheDocument();
+  });
+
+  it('renders a new header when clicking on add button', async () => {
+    render(
+      <AuthFormTestProvider onSubmit={onSubmit}>
+        <HeaderFields maxHeaders={20} readOnly={false} />
+      </AuthFormTestProvider>
+    );
+    const addBtn = await screen.findByTestId('webhookAddHeaderButton');
+    expect(addBtn).toBeInTheDocument();
```

Comment:
nit: I think this is redundant because `findByTestId` will reject if the element is not found.